src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -181,10 +184,20 @@ AsyncLogWriter* AsyncLogWriter::instance() {
   return _instance;
 }
 
-// write() acquires and releases _io_sem even _buffer is empty.
-// This guarantees all logging I/O of dequeued messages are done when it returns.
+// NOT MT-safe! see the comments in the header file.
```

Comment:
Suggested comment:
// Inserts a flush token into the async output buffer and waits until the AsyncLog thread
// signals that it has seen it and completed all existing message processing.
// This is method is not MT-safe in itself, but is guarded by another lock in the usual
// usecase - see the comments in the header file for more details.